Moya/Moya.swift

```diff
@@ -126,11 +129,12 @@ public class MoyaProvider<T: MoyaTarget> {
     public func request(token: T, completion: MoyaCompletion) -> Cancellable {
         let endpoint = self.endpoint(token)
         let request = endpointResolver(endpoint: endpoint)
+        let credential = credentialClosure?(token)
         let stubBehavior = self.stubBehavior(token)
 
         switch stubBehavior {
         case .NoStubbing:
-            return sendRequest(request, completion: completion)
+            return sendRequest(request, credential:credential, completion: completion)
         default:
             return stubRequest(request, completion: completion, endpoint: endpoint, stubBehavior: stubBehavior)
```

Comment:
Do you think it could be valuable to use the credentials when stubbing a request, too? I can't _really_ see how, but wanted to see if it was something we might want to add.